lib/hanami/router.rb

```diff
@@ -868,7 +869,13 @@ def _redirect(to, code)
     def _params(env, params)
       params ||= {}
       env[PARAMS] ||= {}
-      env[PARAMS].merge!(Rack::Utils.parse_nested_query(env["QUERY_STRING"]))
+
+      if (input = env[Rack::RACK_INPUT]) and input.rewind
```

Comment:
Is the `and` here instead of `&&` intentional? I can see how it signifies lower precedence than the assignment in the parens, so just wondering

Reply:
@jodosha Yes, it's intentional. To me reads: "if rack input is present, then rewind it, and enter the if statement".